### DISCUSSION TITLE
Device status to include Decommissioned

### DISCUSSION THREAD

**Discussion by digital-fortress** [netbox-community/netbox#12855] [Ideas]
After reviewing the discussion and the fix regarding the inclusion of "Decommissioning" (#3103), I am considering requesting the addition of a new status value - "Decommissioned."

Based on my understanding, the current status options are as follows:

Active: Represents devices that are currently in use and operational.
Offline: Refers to devices that have been powered off but can be reactivated, typically applicable to servers.
Inventory: Signifies devices that have been decommissioned from a particular site but are still in inventory and can be repurposed for another site or project.
Decommissioning: Devices that are in the process of being decommissioned.
I believe introducing the "Decommissioned" status would improve device categorization by clearly identifying devices that are no longer in active use and are now part of our historical records. This status would apply to devices that have been taken offline, cannot be reused due to being at end-of-life (EOL), and are considered obsolete.

I value your thoughts and feedback on this proposal.

**Comment by candlerb** [comment]
Since Netbox [v3.2.0](https://github.com/netbox-community/netbox/releases/v3.2.0), device status values are now customizable (as well as a number of other status values): see https://docs.netbox.dev/en/stable/configuration/data-validation/#field_choices

Therefore, if you would find this additional choice useful, you are free to add it - after upgrading to v3.2+ if you are on an older version.

**Reply by digital-fortress** [reply]
Thanks for the response. But I am totally confused where this data should be entered.

**Reply by candlerb** [reply]
Please see https://docs.netbox.dev/en/stable/configuration/

Some settings are "dynamic configuration parameters" which you can set via the web interface using Admin > Extras > "config revisions", but `FIELD_CHOICES` isn't one of them.  You have to add it manually to `/opt/netbox/netbox/netbox/configuration.py`.

**Reply by digital-fortress** [reply]
Thank you. I did that and is working. Wow, was so easy in the end. Many thanks again for your help. 

FIELD_CHOICES = {
    'dcim.Device.status+': (
        ('decommissioned', 'Decommissioned', 'yellow'),
    )
}

**Reply by bitcollector1** [reply]
> Since Netbox [v3.2.0](https://github.com/netbox-community/netbox/releases/v3.2.0), device status values are now customizable (as well as a number of other status values): see https://docs.netbox.dev/en/stable/configuration/data-validation/#field_choices
> 
> Therefore, if you would find this additional choice useful, you are free to add it - after upgrading to v3.2+ if you are on an older version.

Wow, this is great info!  It's hard to stay on top of ALL the changes and this tip is especially helpful!
